validator/client/src/test/java/tech/pegasys/teku/validator/client/loader/ExternalValidatorSourceTest.java

```diff
@@ -110,9 +102,53 @@ void shouldThrowExceptionWhenAddValidator() {
   }
 
   @Test
-  void shouldThrowExceptionWhenDeleteValidator() {
-    assertThatThrownBy(() -> validatorSource.deleteValidator(BLSPublicKey.empty()))
-        .isInstanceOf(UnsupportedOperationException.class);
+  void shouldReturnErrorWhenDeleteReadOnlyValidator() {
+    assertThat(validatorSource.deleteValidator(BLSPublicKey.empty()))
+        .isEqualTo(
+            DeleteKeyResult.error(
+                "Cannot delete validator from read-only local validator source."));
+  }
+
+  @Test
+  void shouldDeleteValidator(@TempDir Path tempDir) throws IOException {
+    final BLSPublicKey publicKey = dataStructureUtil.randomPublicKey();
+    final ExternalValidatorSource externalValidatorSource =
+        newExternalValidatorSource(tempDir, false);
+
+    final AddValidatorResult addValidatorResult =
+        externalValidatorSource.addValidator(publicKey, Optional.empty());
+    assertImportedSuccessfully(addValidatorResult, new URL("http://localhost:9000"));
+    assertFileContent(tempDir, publicKey, "{\"pubkey\":\"" + publicKey + "\"}");
+
+    final DeleteKeyResult result = externalValidatorSource.deleteValidator(publicKey);
+    assertThat(result.getStatus()).isEqualTo(DeletionStatus.DELETED);
+    assertThat(result.getMessage()).isEqualTo(Optional.empty());
+
+    final Path path =
+        ValidatorClientService.getManagedRemoteKeyPath(new SimpleDataDirLayout(tempDir))
+            .resolve(publicKey.toBytesCompressed().toUnprefixedHexString() + ".json");
+    assertThat(path.toFile()).doesNotExist();
+  }
+
+  @Test
+  void shouldReturnErrorWhenDeleteNonExistentValidator(@TempDir Path tempDir) {
+    final BLSPublicKey publicKey = dataStructureUtil.randomPublicKey();
+    final ExternalValidatorSource externalValidatorSource =
+        newExternalValidatorSource(tempDir, false);
+
+    final DeleteKeyResult result = externalValidatorSource.deleteValidator(publicKey);
+    assertThat(result.getStatus()).isEqualTo(DeletionStatus.ERROR);
+    assertThat(result.getMessage())
+        .isEqualTo(
```

Comment:
should be `DeletionStatus.NOT_FOUND`

Reply:
I think the distinciton between error and not_found in this case is 
'we had nothing in memory and nothing on disk' == not_found
'we had stuff, and we tried and failed to clean it up' == error (eg. file was read only which should be v. unlikely, but easy to reproduce)

Reply:
equally delete of a readonly should be an error...

Reply:
or a pubkey that's actually present in a different source but not present in the remote read-write - also should be error ...